### PR TITLE
Multithreading tests and diagnostics

### PR DESCRIPTION
**Pull request by sjavora**
Addresses [this FIXME](https://github.com/sjavora/swift-driver/blob/2737f31ad7b598c8a3ed481cad3b96a0c778c5f3/Sources/SwiftDriver/Driver/Driver.swift#L739). Also adds tests for the `// Multithreading` extension.

### CODE REVIEW

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -466,7 +466,7 @@ extension Diagnostic.Message {
     .error("-driver-use-frontend-path requires a Swift compiler executable argument")
   }
 
-  static var warning_cannot_multithread_batch_mode: Diagnostic.Message {
+  public static var warning_cannot_multithread_batch_mode: Diagnostic.Message {
```
Comment (DougGregor) [inline Sources/SwiftDriver/Driver/Driver.swift]: Can we leave these internal and test for the text string? It's repetitive, but we might want to shuffle these things around over time and would rather not commit to them being at all stable.

Comment (sjavora) [reply Sources/SwiftDriver/Driver/Driver.swift]: Done. I left `error_invalid_arg_value` alone, though, because that was already public.

Comment (DougGregor) [reply Sources/SwiftDriver/Driver/Driver.swift]: Thank you! We can update error_invalid_arg_value in another PR

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by sjavora** [comment]
`
/home/buildnode/jenkins/workspace/pr-swift-driver-linux/branch-master/swift-driver/Tests/SwiftDriverTests/SwiftDriverTests.swift:169: error: SwiftDriverTests.testMultithreading : XCTAssertEqual threw error "unableToFind(tool: "clang")" -
`

Not sure what's going on here - the tests don't do anything special 🤔

**Comment by DougGregor** [comment]
Hmm. Perhaps you need to add an explicit `-target` to your `Driver` invocations, e.g., as in nhttps://github.com/apple/swift-driver/blob/master/Tests/SwiftDriverTests/SwiftDriverTests.swift#L232

**Comment by harlanhaskins** [comment]
Every time I've run it on Linux, `clang` has been in my `PATH`. Is it not in the `PATH` on the bots?

Also, we really need to figure out toolchain-relative lookups :/

**Comment by DougGregor** [comment]
> Every time I've run it on Linux, `clang` has been in my `PATH`. Is it not in the `PATH` on the bots?

> Also, we really need to figure out toolchain-relative lookups :/

Yes, definitely.

**Comment by sjavora** [comment]
> https://github.com/apple/swift-driver/blob/master/Tests/SwiftDriverTests/SwiftDriverTests.swift#L232

@DougGregor so this should work? I only have access to a Mac at the moment so I can't really test this. Also, the test failed only on the last assert - should I add the target only there or to all the invocations?

**Comment by sjavora** [comment]
Added the target argument to the one failing assert for now.

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci test

**Comment by sjavora** [comment]
@DougGregor now it failed on the other test - added target there as well, hopefully this will finally pass. I'm sorry about all the hassle for such small changes 😞

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
No worries; we need to figure out how best to find tools within our toolchain so these tests will become target-agnostic, but that's a separate issue.

**Comment by sjavora** [comment]
@DougGregor thanks! Should I open a PR to make some other diagnostics internal? If so, which ones? 

Better yet, is there a place to discuss this project on the Swift forums? https://forums.swift.org/c/development/tooling?

**Comment by DougGregor** [comment]
Yes, I think I'd like to make all of the diagnostics internal. For discussions, I recommend https://forums.swift.org/c/development/compiler

**Pull request merged**